cli/parse-options.js

```diff
@@ -38,8 +38,39 @@ function parseOptions (options) {
     config.url = options.url
   }
 
+  PouchDB.plugin(require('pouchdb-mapreduce'))
+
   if (options.dbUrl) {
-    config.db.url = options.dbUrl
+    config.PouchDB = config.PouchDB.defaults({prefix: options.dbUrl})
+
+    if (!options.db.url) {
+      if (options.inMemory) {
+        PouchDB.plugin(require('pouchdb-adapter-memory'))
+        log.info('config', 'Storing all data in memory only')
+      } else {
+        PouchDB.plugin(require('pouchdb-adapter-leveldb'))
+
+       // this is a temporary workaround until we replace options.db with options.PouchDB:
+       // https://github.com/hoodiehq/hoodie/issues/555
+        if (!options.paths.data) {
+          paths.data = '.hoodie'
+        }
+
+        options.db.prefix = path.join(options.paths.data, 'data' + path.sep)
+        log.info('config', 'No CouchDB URL provided, falling back to PouchDB')
+        log.info('config', 'Writing PouchDB database files to ' + options.db.prefix)
+      }
+    }
+
+    if (options.db.url) {
+      if (!urlParse(options.db.url).auth) {
+        return next(new Error('Authentication details missing from database URL: ' + options.db.url))
+      }
+
+      PouchDB.plugin(require('pouchdb-adapter-http'))
+      options.db.prefix = options.db.url
+      delete options.db.url
+    }
```

Comment:
define the `dbOptions` variable and use it within the `parseOptions` function only. You can add it right on top

```js
function parseOptions (options) {
  var dbConfig = {}
  // ...
```

I would gather `dbConfig` based on the options (e.g. if `options.dbUrl` is set or `options.inMemory` and then at the at set `config.PouchDB = PouchDB.defaults(dbOptions)`

makes sense?